### ISSUE TITLE
Adapter crashing since Hue Bridge Update 1.59.1959097030

### ISSUE THREAD

**Issue by marvdevlabs** [iobroker-community-adapters/ioBroker.hue#445]
Since the update (1.59.1959097030) of the Philips Hue Bridge, the adapter keeps crashing and does not run smoothly.

`
hue.0 | 2023-07-30 15:52:30.425 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
-- | -- | -- | --
hue.0 | 2023-07-30 15:52:30.424 | info | terminating
hue.0 | 2023-07-30 15:52:30.424 | info | cleaned everything up...
hue.0 | 2023-07-30 15:52:30.373 | error | Unexpected token < in JSON at position 0
hue.0 | 2023-07-30 15:52:30.373 | error | SyntaxError: Unexpected token < in JSON at position 0 at JSON.parse (<anonymous>) at uuidResourceParser (/opt/iobroker/node_modules/hue-push-client/dist/index.js:94:21) at IncomingMessage.<anonymous> (/opt/iobroker/node_modules/hue-push-client/dist/index.js:43:36) at IncomingMessage.emit (node:events:525:35) at IncomingMessage.emit (node:domain:489:12) at endReadableNT (node:internal/streams/readable:1358:12) at processTicksAndRejections (node:internal/process/task_queues:83:21)
hue.0 | 2023-07-30 15:52:30.371 | error | unhandled promise rejection: Unexpected token < in JSON at position 0
hue.0 | 2023-07-30 15:52:30.370 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
hue.0 | 2023-07-30 15:52:30.175 | info | Push connection established
hue.0 | 2023-07-30 15:52:25.148 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 15:52:24.770 | info | Push connection established
hue.0 | 2023-07-30 15:52:19.746 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 15:52:19.247 | info | Push connection established
hue.0 | 2023-07-30 15:52:14.227 | info | Push connection established
hue.0 | 2023-07-30 15:52:14.223 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 15:52:13.746 | info | starting. Version 3.9.3 in /opt/iobroker/node_modules/iobroker.hue, node: v16.20.1, js-controller: 4.0.24
`

**Comment by ghost** [comment]
I have the same problem. The adapter connects briefly and immediately goes to "red" and I get a crash message that the adapter crashes frequently.

`<html><body>


hue.0 | 5802 | 2023-07-30 18:17:36.734 | error | Unexpected token < in JSON at position 0
-- | -- | -- | -- | --
hue.0 | 5802 | 2023-07-30 18:17:36.733 | error | SyntaxError: Unexpected token < in JSON at position 0     at JSON.parse (<anonymous>)     at uuidResourceParser (/opt/iobroker/node_modules/hue-push-client/dist/index.js:94:21)     at IncomingMessage.<anonymous> (/opt/iobroker/node_modules/hue-push-client/dist/index.js:43:36)     at IncomingMessage.emit (node:events:525:35)     at IncomingMessage.emit (node:domain:489:12)     at endReadableNT (node:internal/streams/readable:1359:12)     at processTicksAndRejections (node:internal/process/task_queues:82:21)
hue.0 | 5802 | 2023-07-30 18:17:36.725 | error | unhandled promise rejection: Unexpected token < in JSON at position 0
hue.0 | 5802 | 2023-07-30 18:17:36.725 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
hue.0 | 5802 | 2023-07-30 18:17:36.712 | error | Error
hue.0 | 5802 | 2023-07-30 18:17:36.734 | error | Unexpected token < in JSON at position 0
hue.0 | 5802 | 2023-07-30 18:17:36.733 | error | SyntaxError:  Unexpected token < in JSON at position 0    at JSON.parse  (<anonymous>)    at uuidResourceParser  (/opt/iobroker/node_modules/hue-push-client/dist/index.js:94:21)    at  IncomingMessage.<anonymous>  (/opt/iobroker/node_modules/hue-push-client/dist/index.js:43:36)    at  IncomingMessage.emit (node:events:525:35)    at IncomingMessage.emit  (node:domain:489:12)    at endReadableNT  (node:internal/streams/readable:1359:12)    at processTicksAndRejections  (node:internal/process/task_queues:82:21)
hue.0 | 5802 | 2023-07-30 18:17:36.725 | error | unhandled promise rejection: Unexpected token < in JSON at position 0
hue.0 | 5802 | 2023-07-30 18:17:36.725 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
hue.0 | 5802 | 2023-07-30 18:17:36.712 | error | Error
hue.0 | 5893 | 2023-07-30 18:18:49.698 | info | terminating
-- | -- | -- | -- | --
hue.0 | 5893 | 2023-07-30 18:18:49.498 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
hue.0 | 5893 | 2023-07-30 18:18:49.497 | debug | Plugin sentry destroyed
hue.0 | 5893 | 2023-07-30 18:18:49.496 | info | terminating
hue.0 | 5893 | 2023-07-30 18:18:49.495 | info | cleaned everything up...

</body>
</html>
`

Node.js -  v18.16.1
JS_Adapter - 5.0.10
Admin_Adapter - 6.6.4
HUE_Version - 3.9.3

**Comment by foxriver76** [comment]
Hm not sure if it intended, that they removed this endpoint, their docs are currently not available https://developers.meethue.com/

I will make a new version which will at least run in general, if there is not more which they have removed.

**Comment by marvdevlabs** [comment]
thank you very much :)

**Comment by foxriver76** [comment]
Please try 3.9.4 should be on npm in a few minutes

**Comment by marvdevlabs** [comment]
Unfortunately, there are still issues. However, there are different log outputs for SSL and non-SSL.

without SSL
`
hue.0 | 2023-07-30 19:26:42.084 | info | Push connection error: Too Many Requests
-- | -- | -- | --
hue.0 | 2023-07-30 19:26:37.065 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:36.706 | info | Push connection error: Too Many Requests
hue.0 | 2023-07-30 19:26:31.690 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:31.339 | info | Push connection error: Too Many Requests
hue.0 | 2023-07-30 19:26:26.321 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:25.977 | info | Push connection error: Too Many Requests
hue.0 | 2023-07-30 19:26:20.958 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:20.765 | error | Could not get UUIDs: Unexpected token < in JSON at position 0
hue.0 | 2023-07-30 19:26:20.592 | info | Push connection established
hue.0 | 2023-07-30 19:26:15.566 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:15.146 | info | Push connection established
hue.0 | 2023-07-30 19:26:10.122 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:09.787 | info | Push connection established
hue.0 | 2023-07-30 19:26:04.769 | info | Push connection established
hue.0 | 2023-07-30 19:26:04.763 | warn | Could not get configuration from HUE bridge (192.168.1.21:80)
hue.0 | 2023-07-30 19:26:04.272 | info | starting. Version 3.9.4 (non-npm: iobroker-community-adapters/ioBroker.hue#96267297ee822ac1a10b0608c1f8bd0e12dbc359) in /opt/iobroker/node_modules/iobroker.hue, node: v18.17.0, js-controller: 4.0.24

`

with SSL

`

hue.0 | 2023-07-30 19:26:00.326 | info | Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
-- | -- | -- | --
hue.0 | 2023-07-30 19:26:00.325 | info | terminating
hue.0 | 2023-07-30 19:26:00.325 | info | cleaned everything up...
hue.0 | 2023-07-30 19:26:00.300 | info | Got terminate signal TERMINATE_YOURSELF
hue.0 | 2023-07-30 19:25:55.901 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:55.900 | error | Request failed with status code 429
hue.0 | 2023-07-30 19:25:50.800 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:50.800 | error | Request failed with status code 429
hue.0 | 2023-07-30 19:25:45.701 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:45.701 | error | Request failed with status code 429
hue.0 | 2023-07-30 19:25:40.601 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:40.601 | error | Request failed with status code 429
hue.0 | 2023-07-30 19:25:35.495 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:35.494 | error | Request failed with status code 429
hue.0 | 2023-07-30 19:25:30.400 | error | Could not get UUIDs: Unexpected token < in JSON at position 0
hue.0 | 2023-07-30 19:25:30.388 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:30.388 | error | Error
hue.0 | 2023-07-30 19:25:30.386 | info | Push connection established
hue.0 | 2023-07-30 19:25:24.793 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:24.583 | info | Push connection established
hue.0 | 2023-07-30 19:25:18.813 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:18.568 | info | Push connection established
hue.0 | 2023-07-30 19:25:12.906 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-30 19:25:12.499 | info | Push connection established
hue.0 | 2023-07-30 19:25:11.941 | info | starting. Version 3.9.4 (non-npm: iobroker-community-adapters/ioBroker.hue#96267297ee822ac1a10b0608c1f8bd0e12dbc359) in /opt/iobroker/node_modules/iobroker.hue, node: v18.17.0, js-controller: 4.0.24

`

**Comment by marvdevlabs** [comment]
and this with SSL and polling on 

`

hue.0 | 2023-07-31 08:16:54.700 | info | Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
-- | -- | -- | --
hue.0 | 2023-07-31 08:16:54.700 | info | terminating
hue.0 | 2023-07-31 08:16:54.699 | info | cleaned everything up...
hue.0 | 2023-07-31 08:16:54.694 | info | Got terminate signal TERMINATE_YOURSELF
hue.0 | 2023-07-31 08:16:52.379 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:52.379 | error | Request failed with status code 429
hue.0 | 2023-07-31 08:16:50.424 | error | Could not poll all: Error
hue.0 | 2023-07-31 08:16:47.278 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:47.278 | error | Request failed with status code 429
hue.0 | 2023-07-31 08:16:42.181 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:42.181 | error | Request failed with status code 429
hue.0 | 2023-07-31 08:16:40.409 | error | Could not poll all: Error
hue.0 | 2023-07-31 08:16:37.083 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:37.083 | error | Request failed with status code 429
hue.0 | 2023-07-31 08:16:31.984 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:31.984 | error | Request failed with status code 429
hue.0 | 2023-07-31 08:16:30.401 | error | Could not poll all: Error
hue.0 | 2023-07-31 08:16:26.894 | error | Could not get UUIDs: Unexpected token < in JSON at position 0
hue.0 | 2023-07-31 08:16:26.882 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:26.882 | error | Error
hue.0 | 2023-07-31 08:16:26.880 | info | Push connection established
hue.0 | 2023-07-31 08:16:21.466 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:21.244 | info | Push connection established
hue.0 | 2023-07-31 08:16:20.391 | error | Could not poll all: Cannot convert undefined or null to object
hue.0 | 2023-07-31 08:16:15.631 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:15.428 | info | Push connection established
hue.0 | 2023-07-31 08:16:10.165 | error | Could not poll all: Cannot convert undefined or null to object
hue.0 | 2023-07-31 08:16:09.820 | warn | Could not get configuration from HUE bridge (192.168.1.21:443)
hue.0 | 2023-07-31 08:16:09.400 | info | Push connection established
hue.0 | 2023-07-31 08:16:08.819 | info | starting. Version 3.9.4 (non-npm: iobroker-community-adapters/ioBroker.hue#96267297ee822ac1a10b0608c1f8bd0e12dbc359) in /opt/iobroker/node_modules/iobroker.hue, node: v18.17.0, js-controller: 4.0.24

`

**Comment by senger1985** [comment]
same problem here

**Comment by marvdevlabs** [comment]
> Hm not sure if it intended, that they removed this endpoint, their docs are currently not available https://developers.meethue.com/
> 
> I will make a new version which will at least run in general, if there is not more which they have removed.

@foxriver76 do you have new informations ?

**Comment by foxriver76** [comment]
How high is your polling interval configured?

not not really changelog stating nothing about restrictions for the new version

**Comment by marvdevlabs** [comment]
10 seconds, default Setting

Max Hauser ***@***.***> schrieb am Di. 1. Aug. 2023 um 18:50:

**Comment by rof999** [comment]
I have installed Release 3.9.5, no crashes since the installation.

**Comment by marvdevlabs** [comment]
Okay, that's curious. What settings do you have in the adapter? I always
get an error that it can't fetch data from the bridge, but I can access the
API and control it using the IP address without any issues. rof999 <
***@***.***> schrieb am Di. 1. Aug. 2023 um 20:44:

**Comment by foxriver76** [comment]
hm okay, because according to the code it would mean `too many requests`. Do you have any other application maybe a second hue adapter accessing the same bridge? Or a script with an infinite loop maybe?

**Comment by marvdevlabs** [comment]
No, this is the only Hue adapter, and nothing else is accessing the Hue Bridge.

Here is the current log; apparently, it still has issues with the UUIDs. I've also tried deleting the adapter, deleting the "Hue" folder, deactivating JavaScript adapter, and scripts that access Hue lamps. I suspected it could be an issue with the Hue Bridge. However, since I can access it without any problems using the app and Alexa to control the lamps normally, I would rule that out.

PS: ist es eigentlich Vorschrift auf englisch zu schreiben ? :D 


`

2023-08-02 07:10:26.667 - info: hue.0 (307276) starting. Version 3.9.5 (non-npm: iobroker-community-adapters/ioBroker.hue#48b650de57e901971218426c3009b1da74c41d1b) in /opt/iobroker/node_modules/iobroker.hue, node: v18.17.0, js-controller: 4.0.24
--
2023-08-02 07:10:27.166 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:27.170 - info: hue.0 (307276) Push connection established
2023-08-02 07:10:27.374 - error: hue.0 (307276) Could not poll all: Cannot convert undefined or null to object
2023-08-02 07:10:32.532 - info: hue.0 (307276) Push connection established
2023-08-02 07:10:32.535 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:37.938 - info: hue.0 (307276) Push connection established
2023-08-02 07:10:37.940 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:43.331 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:43.332 - info: hue.0 (307276) Push connection established
2023-08-02 07:10:43.346 - error: hue.0 (307276) Could not get UUIDs: Unexpected token < in JSON at position 0
2023-08-02 07:10:47.557 - error: hue.0 (307276) Could not poll all: Cannot convert undefined or null to object
2023-08-02 07:10:48.349 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:10:48.695 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:53.715 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:10:54.062 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:10:59.079 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:10:59.427 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:04.445 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:04.803 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:07.738 - error: hue.0 (307276) Could not poll all: Cannot convert undefined or null to object
2023-08-02 07:11:09.822 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:10.174 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:15.192 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:15.540 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:20.556 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:21.078 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:26.094 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:26.446 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:27.917 - error: hue.0 (307276) Could not poll all: Cannot convert undefined or null to object
2023-08-02 07:11:31.464 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:31.792 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:36.809 - info: hue.0 (307276) Push connection error: Too Many Requests
2023-08-02 07:11:37.131 - warn: hue.0 (307276) Could not get configuration from HUE bridge (192.168.1.21:80)
2023-08-02 07:11:38.127 - info: host.ioBroker "system.adapter.hue.0" disabled
2023-08-02 07:11:38.127 - info: host.ioBroker stopInstance system.adapter.hue.0 (force=false, process=true)
2023-08-02 07:11:38.129 - info: hue.0 (307276) Got terminate signal TERMINATE_YOURSELF
2023-08-02 07:11:38.132 - info: hue.0 (307276) cleaned everything up...
2023-08-02 07:11:38.132 - info: hue.0 (307276) terminating
2023-08-02 07:11:38.133 - info: hue.0 (307276) Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
2023-08-02 07:11:38.177 - info: host.ioBroker stopInstance system.adapter.hue.0 send kill signal
2023-08-02 07:11:38.671 - info: host.ioBroker instance system.adapter.hue.0 terminated with code 11 (ADAPTER_REQUESTED_TERMINATION)


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">`
